### PR TITLE
[WIP] implement dispose method for data source

### PR DESCRIPTION
**Pull request by 5ebastianMeier**
Depends on https://github.com/essential-projects/data_model_contracts/pull/15

Closes https://github.com/essential-projects/data_model/issues/13

## What did you change?

* Implemented the `dispose` method on `EntityType`, `DataModel` and `DataSource`

## How can others test the changes?

See https://github.com/process-engine/process_engine_meta/pull/72

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/data_model.ts [hunk 1]
```diff
@@ -59,6 +59,17 @@ export class DataModel<T extends IEntity> implements IDataModel<T> {
     return this._entityTypeCache;
   }
 
+  public async dispose(): Promise<void> {
+    for (const configHash in this.entityTypeCache) {
```
Comment (cmg-dev) [inline src/data_model.ts]: _entityTypeCache

File: src/data_model.ts [hunk 2]
```diff
@@ -59,6 +59,17 @@ export class DataModel<T extends IEntity> implements IDataModel<T> {
...
+      const configEntityTypes: any = this.entityTypeCache[configHash];
```
Comment (cmg-dev) [inline src/data_model.ts]: _entityTypeCache

File: src/data_source.ts [hunk 3]
```diff
@@ -45,6 +45,10 @@ export class DataSource<T extends IEntity> implements IDataSource<T> {
     this._adapter = await this._createAdapter();
   }
 
+  public async dispose(): Promise<void> {
+    return this.adapter.dispose();
```
Comment (cmg-dev) [inline src/data_source.ts]: `this._adapter`

please overhaul all private member accesses.

**Comment by cmg-dev** [comment]
here should be a little bit of refactoring neccessary.

E.g. the config is public... this should not be the case. and if this is necessary: we need to document it!

**Comment by ElRaptorus** [comment]
This is going to be included in the next major release.

**Pull request closed without merging**